Improvements for AKS documentation (#2442)
kyma-project/kyma: docs/kyma/docs/04-05-aks-installation.md

### Environment variables

Set the following environment variables:
* To set mandatory variables, run:
  ```
  export RS_GROUP={YOUR_RESOURCE_GROUP_NAME}
  export CLUSTER_NAME={YOUR_CLUSTER_NAME}
  export REGION={YOUR_REGION} #westeurope
  ```

> * If you want to install from release go to [this](https://github.com/kyma-project/kyma/releases/) page and choose the release you want to use.

Reviewer: ```suggestion
* If you want to install Kyma from release,  choose a [release](https://github.com/kyma-project/kyma/releases/) you want to use and export it as an environment variable. Run:
```

>   Export the version you chose as an environment variable. Run:

Reviewer: ```suggestion

```

  ```
  export KYMA_RELEASE_VERSION={KYMA_RELEASE_VERSION}
  ```
  >**NOTE:** Installation from release is supported since version 0.7.0. 

>   If you want to install from sources do not export `KYMA_RELEASE_VERSION`.

Reviewer: ```suggestion
  If you want to install Kyma from sources, do not export `KYMA_RELEASE_VERSION`.
```

    az aks get-credentials --resource-group $RS_GROUP --name $CLUSTER_NAME
    ```

3. Install Tiller and add additional privileges to be able to access readiness probes endpoints on your AKS cluster.
    * Installation from release
    ```
    kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/$KYMA_RELEASE_VERSION/installation/resources/tiller.yaml
    kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/$KYMA_RELEASE_VERSION/installation/resources/azure-crb-for-healthz.yaml
    ```

>     * If you are installing from sources, checkout [kyma-project](https://github.com/kyma-project/kyma) and enter the root folder. Run:

Reviewer: ```suggestion
    * If you install Kyma from sources, check out [kyma-project](https://github.com/kyma-project/kyma) and enter the root folder. Run:
```

    ```

2. Push the image to your Docker Hub:
    ```
    docker tag kyma-installer:latest {YOUR_DOCKER_LOGIN}/kyma-installer:latest
    ```
    ```
    docker push {YOUR_DOCKER_LOGIN}/kyma-installer:latest
    ```

3. Prepare the deployment file:

    ```
    cat installation/resources/installer.yaml <(echo -e "\n---") installation/resources/installer-config-cluster.yaml.tpl  <(echo -e "\n---") installation/resources/installer-cr-cluster.yaml.tpl | sed -e "s/__PROXY_EXCLUDE_IP_RANGES__/10.0.0.1/g" | sed -e "s/__DOMAIN__/$SUB_DOMAIN.$DNS_DOMAIN/g" |sed -e "s/__TLS_CERT__/$TLS_CERT/g" | sed -e "s/__TLS_KEY__/$TLS_KEY/g" | sed -e "s/__.*__//g" > my-kyma.yaml
    ```

4. In the `my-kyma.yaml` file created in the previous step, change the image URL to `{YOUR_DOCKER_LOGIN}/kyma-installer:latest`.
    ```
    kind: Deployment
    metadata:
      name: kyma-installer
      namespace: kyma-installer
      ......
        image: eu.gcr.io/kyma-project/develop/installer:30bf314d
    ```
    Use the modified file to deploy Kyma on your AKS cluster.

## Deploy Kyma

1. Deploy Kyma using the `my-kyma` custom configuration file you created. Run:
    ```
    kubectl apply -f my-kyma.yaml
    ```

>     >**NOTE:** In case you receive an error: `Error from server (MethodNotAllowed)` run the command again before going to point no 2.

Reviewer: ```suggestion
    >**NOTE:** In case you receive the `Error from server (MethodNotAllowed)` error, run the command again before going to step 2.
```

    
2. Check if the Pods of Tiller and the Kyma Installer are running:
    ```
    kubectl get pods --all-namespaces
    ```

3. Start Kyma installation:
    ```
    kubectl label installation/kyma-installation action=install
    ```

4. To watch the installation progress, run:
    ```
    while true; do \
      kubectl get installation/kyma-installation -o jsonpath="{'Status: '}{.status.state}{', description: '}{.status.description}{'\n'}"; \
      sleep 5; \
    done
    ```

>     When installation process is finished you will see a `Status: Installed, description: Kyma installed` message.

Reviewer: ```suggestion
    After the installation process is finished, the `Status: Installed, description: Kyma installed` message appears.
```

>     In case of error you can fetch the logs from the installer by running:

Reviewer: ```suggestion
    In case of an error, you can fetch the logs from the Installer by running:
```
